core/utility/src/main/java/com/waminiyi/utility/utils/NetworkUtils.kt

```diff
@@ -0,0 +1,29 @@
+package com.waminiyi.utility.utils
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+/**
+ * Class for useful network functions
+ */
+class NetworkUtils @Inject constructor(@ApplicationContext private val context: Context) {
```

Comment:
Tu devrais injecter ConnectivityManager plutot que context
```suggestion
class NetworkUtils @Inject constructor(private val connectivityManager: ConnectivityManager) {
```

Et tu crée un hilt module pour ca
```kotlin
@Provides
fun provideConnectivityManager(@ApplicationContext private val context: Context): ConnectivityManager =
  context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
```